### PR TITLE
doc/user: update documentation to account for multi-output sources

### PR DESCRIPTION
**Pull request by morsapaes**
Adapting existing documentation that touches on Postgres and load generator sources to the changes in #14016.

<hr>

**Questions:** 

* Is there a way to correlate a subsource to its parent source using the stable system catalog? At first glance, `SHOW SOURCES` can get pretty wild pretty quickly without some sort of `source->subsource` mapping. It's also confusing to understand the hierarchy (or just infer what type of subsource we're looking at IWRT the source system) by just looking at system tables:

  ```sql
  materialize=> SELECT * FROM mz_sources WHERE name IN ('auction_house','bids');

    id  |  oid  | schema_id |     name      |      type      | connection_id |  size
  ------+-------+-----------+---------------+----------------+---------------+---------
   u174 | 34108 | 3         | bids          | subsource      |               | 3xsmall
   u175 | 34109 | 3         | auction_house | load-generator |               | 3xsmall

  materialize=> SELECT * FROM mz_objects WHERE oid IN (34108,34109);

    id  |  oid  | schema_id |     name      |  type
  ------+-------+-----------+---------------+--------
   u174 | 34108 | 3         | bids          | source
   u175 | 34109 | 3         | auction_house | source
  ```

* Should subsources be associated with a `size`? If ingestion is happening via the parent source, I'd expect them not to! @sjwiesman raised the same question [here](https://materializeinc.slack.com/archives/C0450R1BYBU/p1665617861904239).

### CODE REVIEW

File: doc/user/content/sql/create-source/load-generator.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ _src_name_  | The name for the source.
 **AUCTION** | Use the [auction](#auction) load generator.
 **IF NOT EXISTS**  | Do nothing (except issuing a notice) if a source with the same name already exists.
 **TICK INTERVAL**  | The interval at which the next datum should be emitted. Defaults to one second.
+**FOR ALL TABLES** | Creates subsources for all data sources in the load generator.
+**FOR TABLES** _table_name_ | Creates subsources for specific data sources in the load generator.
```
Comment (benesch) [inline doc/user/content/sql/create-source/load-generator.md:34]: Should be added for PostgreSQL too right?

**Review by benesch — COMMENTED** [review]
I couldn't tell from your Slack message whether you meant that you had uncommitted changes to postgres.md locally that you hadn't pushed up? The stuff you've pushed up so far LGTM though.

**Comment by morsapaes** [comment]
I do, @benesch! Just want to run through the steps and see how things look like before pushing the changes to the source page and the integration guides in.

File: doc/user/content/sql/create-source/load-generator.md (hunk 1 above)
Comment (morsapaes) [reply doc/user/content/sql/create-source/load-generator.md:34]: Yeah!

File: doc/user/content/sql/create-source/load-generator.md [hunk 2]
```diff
@@ -99,7 +101,7 @@ views:
     `amount`     | [`bigint`]                   | The bid amount in dollars.
     `bid_time`   | [`timestamp with time zone`] | The time at which the bid was placed.
 
-The organizations, users, and accounts are fixed at the time the auction source
+The `organizations`, `users`, and `accounts` are fixed at the time the source
```
Comment (benesch) [inline doc/user/content/sql/create-source/load-generator.md:104]: nit: I wrote this meaning the non-monospace organizations—i.e., the actual organizations entities, not the table itself. If you want to switch to refer to the subsources I think you either need to drop the `The` or add `subsources`:

```suggestion
`organizations`, `users`, and `accounts` are fixed at the time the source
```

```suggestion
The `organizations`, `users`, and `accounts` subsources are fixed at the time the source
```

File: doc/user/content/sql/create-source/postgres.md [hunk 3]
```diff
@@ -45,70 +47,46 @@ For this reason, the upstream database must be configured to support logical rep
...
-  (PUBLICATION 'mz_source');
+  FROM POSTGRES
+    CONNECTION pg_connection
+    (PUBLICATION 'mz_source')
+  FOR ALL TABLES
+  WITH (SIZE '3xsmall');
 ```
 
-, its schema looks like:
+, Materialize will automatically create a **subsource** for each original table in the publication:
```
Comment (benesch) [inline doc/user/content/sql/create-source/postgres.md:61]: This leading comma does not look right to my eye:

image


```suggestion
Materialize will automatically create a **subsource** for each original table in the publication:
```

File: doc/user/content/sql/create-source/postgres.md [hunk 4]
```diff
@@ -170,9 +148,11 @@ CREATE CONNECTION pg_connection
...
 ```sql
 CREATE SOURCE mz_source
-FROM POSTGRES
-  CONNECTION pg_connection
-  (PUBLICATION 'mz_source');
+  FROM POSTGRES
+    CONNECTION pg_connection
+    (PUBLICATION 'mz_source')
+  FOR ALL TABLES
+  WITH ( SIZE = 'xsmall' );
```
Comment (benesch) [inline doc/user/content/sql/create-source/postgres.md:155]:
```suggestion
  WITH (SIZE = 'xsmall');
```

File: doc/user/content/sql/create-source/postgres.md [hunk 5]
```diff
@@ -170,9 +148,11 @@ CREATE CONNECTION pg_connection
 
 ```sql
 CREATE SOURCE mz_source
-FROM POSTGRES
-  CONNECTION pg_connection
-  (PUBLICATION 'mz_source');
+  FROM POSTGRES
+    CONNECTION pg_connection
+    (PUBLICATION 'mz_source')
```
Comment (benesch) [inline doc/user/content/sql/create-source/postgres.md:153]:
```suggestion
    CONNECTION pg_connection (PUBLICATION 'mz_source')
```

I think we should try to consistently format these like `CREATE TABLE`, either all one line, or with the open parens starting an indented block:

```
CONNECTION pg_connection (
    PUBLICATION 'mz_source',
    IDKSOMEOTHEROPTION 'we add someday'
)
```

File: doc/user/content/sql/create-source/postgres.md [hunk 6]
```diff
@@ -181,10 +161,10 @@ To provision a specific amount of CPU and memory to a source on creation, use th
 
 ```sql
 CREATE SOURCE mz_source
-FROM POSTGRES
-  CONNECTION pg_connection
-  (PUBLICATION 'mz_source')
-  WITH (SIZE = 'xsmall');
+  FROM POSTGRES
+    CONNECTION pg_connection
+    (PUBLICATION 'mz_source')
```
Comment (benesch) [inline doc/user/content/sql/create-source/postgres.md:166]:
```suggestion
    CONNECTION pg_connection (PUBLICATION 'mz_source')
```

File: doc/user/content/sql/create-source/postgres.md [hunk 7]
```diff
@@ -45,70 +47,46 @@ For this reason, the upstream database must be configured to support logical rep
...
 
 ```sql
-CREATE VIEWS FROM SOURCE mz_source
-(public.foo AS foo, otherschema.foo AS foo2);
+CREATE SOURCE mz_source
+  FROM POSTGRES
+    CONNECTION pg_connection
+    (PUBLICATION 'mz_source')
+  FOR TABLES ( schema1.table_1 AS s1_table_1,
+               schema2_table_1 AS s2_table_1 )
```
Comment (benesch) [inline doc/user/content/sql/create-source/postgres.md:87]:
```suggestion
  CONNECTION pg_connection (PUBLICATION 'mz_source')
  FOR TABLES (
    schema1.table_1 AS s1_table_1,
    schema2_table_1 AS s2_table_1
  )
```

**Review by benesch — APPROVED** [review]
:yay-fox:

File: doc/user/content/sql/create-source/postgres.md (hunk 5 above)
Comment (morsapaes) [reply doc/user/content/sql/create-source/postgres.md:153]: > IDKSOMEOTHEROPTION 'we add someday'

:blob-no:

Will make them consistently use one line for now.

File: doc/user/content/get-started.md [hunk 8]
```diff
@@ -48,31 +48,27 @@ We'll start with some real-time data produced by Materialize's built-in [load ge
 
     ```sql
     CREATE SOURCE auction_house
-    FROM LOAD GENERATOR AUCTION
-    FOR ALL TABLES;
+        FROM LOAD GENERATOR AUCTION
+        FOR ALL TABLES
+        WITH (SIZE '3xsmall');
```
Comment (uce) [inline doc/user/content/get-started.md:53]: I didn't know that the `=` is optional (e.g. `SIZE = '3xsmall'`). On a quick skim, we've been using the variant with `=` more frequently.

File: doc/user/content/sql/create-source/load-generator.md (hunk 1 above)
Comment (uce) [inline doc/user/content/sql/create-source/load-generator.md:34]:
```suggestion
**FOR ALL TABLES** | Creates subsources for all tables in the load generator.
**FOR TABLES** _table_name_ | Creates subsources for specific tables in the load generator.
```

(optional for consistency)

File: doc/user/content/integrations/cdc-postgres.md [hunk 9]
```diff
@@ -78,42 +78,36 @@ As a _superuser_:
...
-
-_Create views for all tables_
-
-```sql
-CREATE VIEWS FROM SOURCE mz_source;
+CREATE SOURCE mz_source
+  FROM POSTGRES
+    CONNECTION pg_connection (PUBLICATION 'mz_source')
+  FOR TABLES ( table_1,
+               table_2 AS alias_table_2 )
```
Comment (uce) [inline doc/user/content/integrations/cdc-postgres.md:100]: Does Nikhil's formatting feedback apply here, too?

**Review by uce — APPROVED** [review]
LGTM mod inline questions.

* In `get-started.md`, I noticed that we use a non-existing replica size `bigger`.
* I was wondering whether we should mention that you have to DROP the parent source and you can't drop subsources individiually. Do you think that is important to point out explicitly? I think what we have is fine, but wanted to ask nonetheless.

---

Regarding your questions:

1. I agree! @petrosagg has a follow-up planned to make it possible to correlate subsource with their parent.
2. I see how that is confusing. We should add a follow-up to address this, @petrosagg.

**Comment by morsapaes** [comment]
Thanks for the extra set of eyes, @uce @benesch! Just pushed some more edits in, and will merge as is just so that users don't run into stale syntax. There are some things I want to follow-up on, like digging a bit more into the responsibilities of the parent source vs. the subsources, but I'll address that in a separate PR.

**Pull request merged**